app/assets/javascripts/actions/note.js

```diff
@@ -0,0 +1,23 @@
+import actionTypes from './actionTypes';
+
+export const deleteNoteSuccess = (storyId, noteId) => ({
+  type: actionTypes.DELETE_NOTE,
+  storyId,
+  noteId
+});
+
+export const createNoteSuccess = (storyId, note) => ({
+  type: actionTypes.ADD_NOTE,
+  storyId,
+  note
+});
+
+export const deleteNote = (projectId, storyId, noteId) =>
+  (dispatch, getState, { Note }) =>
+    Note.destroy(projectId, storyId, noteId)
+      .then(dispatch(deleteNoteSuccess(storyId, noteId)));
```

Comment:
You need a lambda here in order to make it dispatch just after the promise resolves.